### PR TITLE
Selenium fixes

### LINKED ISSUE #12864: hwinfo, Machines: Crash on Edge with "SCRIPT1003: Expected ':'
Cockpit version: 202
OS: Fedora 30 (host) + Windows 10 (browser)
Page: Machines

The Machines page currently crashes immediately when opening it in Edge (at least the old version on our windows-10 image). This is detected by our 
[selenium-edge test](https://209.132.184.41:8493/logs/pull-12798-20190920-074343-a8d82c89-cockpit-project-cockpit--fedora-30-selenium-edge/log.html), except that we ignored it recently because it's so exceptionally hard to get green.

But this is a genuine crash, not related to cockpit-tls (see PR #12862).

The browser console says

     SCRIPT1003: Expected ':'     machines.js (94749, 4)

Unfortunately, trying to click on this to debug it crashes Edge or doesn't do anything. But when I look at that line in ./dist/machines/machines.js I see
```js
  }) => rowData.hasOwnProperty('parent') && {
    colSpan: colSpan + !!rowData.fullWidth,
    id: contentId + rowIndex,
    className: rowData.noPadding && Object(_patternfly_react_styles__WEBPACK_IMPORTED_MODULE_1__["css"])(_patternfly_react_styles_css_components_Table_table__WEBPACK_IMPORTED_MODULE_2___default.a.modifiers.noPadding)
  };  
```

which looks fine at first sight. We use arrow functions all over the place, and object assignment has been in JS since the beginnig of time.

### PR DESCRIPTION
**Pull request by martinpitt**
This fixes the Edge crashes on hwinfo/Machines pages, and various logic bugs in the Selenium tests.

 - [x] Fix Edge crash on machines/hwinfo page: PR #13056
 - [ ] Some simple Selenium test fixes: PR #13057
 - [ ] Fix data-loaded handling: PR #13059

### CODE REVIEW

**Comment by martinpitt** [comment]
That [latest run](https://209.132.184.41:8493/logs/pull-13052-20191028-220404-e2141bb4-cockpit-project-cockpit--fedora-30-selenium-edge/log.html) looks promising! Let's try again with the Machines tests.

**Comment by martinpitt** [comment]
The latest [run with Machines tests](https://209.132.184.41:8493/logs/pull-13052-20191029-054848-99366a21-cockpit-project-cockpit--fedora-30-selenium-edge/log.html) looks really promising! There are two errors, but specific to Machines. I'll re-disable them for this PR, as I don't want to debug too many issues at once; but after we fix the generic bugs, we should re-attempt that.

FYI @KKoukiou and @skobyda

File: autogen.sh [hunk 1]
```diff
@@ -58,6 +58,11 @@ while ! npm install; do
   find node_modules -mindepth 1 -not -path node_modules/README -delete
 done
 
+# HACK: https://github.com/cockpit-project/cockpit/issues/12864
+# There's something about the ES6 version of PF4's collapsible.js that Edge
+# cannot stomach, so force the ES5 version
+cp node_modules/@patternfly/react-table/dist/js/components/Table/utils/decorators/collapsible.js node_modules/@patternfly/react-table/dist/esm/components/Table/utils/decorators/collapsible.js
```
Comment (KKoukiou) [inline autogen.sh]: Shall we open an issue about that to them?

Comment (martinpitt) [reply autogen.sh]: FTR: This doesn't happen on current Edge, only on our two-year old version on our windows-10 image. Also, this feels a bit like an error on our side: They *do* ship an ES5 plain .js in dist/js, but our build system uses dist/esm/ without running it through babel. I'll soon send a better fix for this.

**Comment by martinpitt** [comment]
So there's still some problems:
 - [firewalld tests have a race condition](https://209.132.184.41:8493/logs/pull-13052-20191029-091010-2067fc85-cockpit-project-cockpit--fedora-30-selenium-edge/log.html) on enabling the unit -- it doesn't wait until it's actually running
 - [kdump race](https://209.132.184.41:8493/logs/pull-13052-20191029-091010-2067fc85-cockpit-project-cockpit--fedora-30-selenium-edge/log.html#23)
 - there's still tests failing on [stale](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-13052-20191029-091010-2067fc85-cockpit-project-cockpit--fedora-30-selenium-firefox/log.html#6) [elements](https://209.132.184.41:8493/logs/pull-13052-20191029-091010-2067fc85-cockpit-project-cockpit--fedora-30-selenium-edge/log.html#35)

**Comment by martinpitt** [comment]
Meh, I was hoping that [this patch](https://github.com/cockpit-project/cockpit/pull/13052/commits/47fdf67f5f09c162804f89ba8fcedd83d36c1fc1) would fix the stale elements, but it [doesn't](https://209.132.184.41:8493/logs/pull-13052-20191029-100909-47fdf67f-cockpit-project-cockpit--fedora-30-selenium-firefox/log.html#8).

**Comment by martinpitt** [comment]
I split out things into separate PRs, as it's just too hairy to fix it all in one.

**Pull request closed without merging**